### PR TITLE
New module postgresql_ping: Checks remote PostgreSQL server availability.

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
New module postgresql_ping: Simple module to check remote PostgreSQL server availability.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
postgresql_ping

##### ANSIBLE VERSION
```
2.8
```

##### RETURNS
```yaml
is_available:
    description: PostgreSQL server availability.
    returned: always
    type: bool
    sample: true

server_version:
    description: PostgreSQL server version.
    returned: always
    type: dict
    sample: { major: 10, minor: 1 }
```

##### EXAMPLES
```yaml
# PostgreSQL ping dbsrv server from the shell:
ansible dbsrv -m postgresql_ping
```
```yaml
# In the example below you need to generate sertificates previously.
# See https://www.postgresql.org/docs/current/libpq-ssl.html for more information.
- name: PostgreSQL ping dbsrv server using not default credentials and ssl
  postgresql_ping:
    db: protected_db
    login_host: dbsrv
    login_user: secret
    login_password: secret_pass
    ssl_rootcert: /root/root.crt
    ssl_mode: verify-full
```

##### OUTPUT EXAMPLES
In following examples Postgres on dbsrv1 is OK, on dbsrv2 it is not running:
```json
dbsrv1 | SUCCESS => {
    "changed": false,
    "is_available": true,
    "server_version": {
        "major": 10, 
        "minor": 6
    }
}
 [WARNING]: PostgreSQL server is unavailable: could not connect to server: No such file or directory         Is the server running locally and accepting         connections
on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?

dbsrv2 | SUCCESS => {
    "changed": false,
    "is_available": false,
    "server_version": {}
}
```

### CODE REVIEW

**Comment by Andersson007** [comment]
@dagwieers , hello. I added my first integration test for this module. It's a very simple module so the test very simple too.
This module don't change anything on the target, therefor I made three check: return values, return values if the database to connect doesn't exist and check mode.
All checks were passed as expected. **Could you look at it once, please?** If it's OK - good and I'm going to continue to make tests for other modules by myself.
Test output example:
```
TASK [postgresql : postgresql_ping - test return values] ***********************
ok: [testhost] => {"changed": false, "database": "postgres", "is_available": true, "port": 5432, "server_version": "10.6", "sslmode": "prefer", "user": "postgres"}

TASK [postgresql : assert] *****************************************************
ok: [testhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [postgresql : postgresql_ping - test check mode (for order, this module doesn't change anything)] ***
ok: [testhost] => {"changed": false, "database": "postgres", "is_available": true, "port": 5432, "server_version": "10.6", "sslmode": "prefer", "user": "postgres"}

TASK [postgresql : assert] *****************************************************
ok: [testhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [postgresql : postgresql_ping - check ping of non-existing database doesn't return anything] ***
ok: [testhost] => {"changed": false, "database": "fake_db", "is_available": false, "port": 5432, "server_version": "", "sslmode": "prefer", "user": "postgres"}

TASK [postgresql : assert] *****************************************************
ok: [testhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}
```

**Review by dagwieers — APPROVED** [review]
I made some cosmetic and docs changes to speed up this process.
LGTM, but I haven't tested the code myself.

**Comment by Andersson007** [comment]
cc @sebasmannem @amenonsen

**Review by sebasmannem — APPROVED** [review]
Hmm, I see some pylint recommendations on variables like e and kw.
I know these examples are out there in other ansible modules, but I would fix them.
Makes it just a little bit cleaner. Also some docstring recommendations.
But I'm okay with shipping it in its current form also.

Generall, I like this module. It will be beneficcial into some crude checking if postgres is up, and init'ed as expected.

**Comment by Andersson007** [comment]
@sebasmannem thank you for the review!

**Comment by dagwieers** [comment]
@sebasmannem As a reviewer you can add a **shipit** to any PR simply by adding a comment/review and mention `shipit` on a new line. A PR requires 2 **shipits** (a green CI check and no change requests) to get merged automagically by @ansibot.

The only exception is a PR that contains a new module (or modifies something outside of the control of the PostgreSQL Working Group). So for this PR a core member still needs to validate the module before merging, but the **shipits** do help understand that it is vetted by maintainers/reviewers.

We currently have [19 (not new_module) PostgreSQL-related pull-requests](https://github.com/ansible/ansible/pulls?utf8=%E2%9C%93&q=is%3Aopen+is%3Apr+label%3Apostgresql+-label%3Anew_module) that need **shipits** from reviewers to get merged. So please help spread the gospel and let those **shipits** count :-)

**Comment by sebasmannem** [comment]
shipit

**Comment by sebasmannem** [comment]
`shipit`

**Review by sebasmannem — APPROVED** [review]
shipit

**Comment by cmfrtl** [comment]
shipit

**Review by kostiantyn-nemchenko — APPROVED** [review]

**Comment by kostiantyn-nemchenko** [comment]
shipit

**Comment by gundalow** [comment]
Thanks everybody, merged into `devel` for release in Ansible 2.8

**Comment by Andersson007** [comment]
Thanks all !

**Pull request merged**
